e2e/nomostest/ntopts/multi_repo.go

```diff
@@ -104,3 +110,54 @@ func WithReconcileTimeout(timeout time.Duration) func(opt *New) {
 		opt.ReconcileTimeout = &timeoutCopy
 	}
 }
+
+// NSReconcilerRBACAdmin specifies to create a NS reconciler ClusterRole
+// which grants rbac admin
+func NSReconcilerRBACAdmin(opt *New) {
+	policy := rbacv1.PolicyRule{
+		APIGroups: []string{rbacv1.GroupName},
+		Resources: []string{rbacv1.ResourceAll},
+		Verbs:     []string{rbacv1.VerbAll},
+	}
+	NSReconcilerCustomPolicy(policy)(opt)
+}
+
+// NSReconcilerCoreAdmin specifies to create a NS reconciler ClusterRole
+// which grants core admin
+func NSReconcilerCoreAdmin(opt *New) {
+	policy := rbacv1.PolicyRule{
+		APIGroups: []string{corev1.GroupName},
+		Resources: []string{rbacv1.ResourceAll},
+		Verbs:     []string{rbacv1.VerbAll},
+	}
+	NSReconcilerCustomPolicy(policy)(opt)
+}
+
+// NSReconcilerRepoSyncAdmin specifies to create a NS reconciler ClusterRole
+// which grants RepoSync admin
+func NSReconcilerRepoSyncAdmin(opt *New) {
+	policy := rbacv1.PolicyRule{
+		APIGroups: []string{configsync.GroupName},
+		Resources: []string{"reposyncs"},
+		Verbs:     []string{rbacv1.VerbAll},
+	}
+	NSReconcilerCustomPolicy(policy)(opt)
+}
+
+// NSReconcilerNamespaceAdmin specifies to create a NS reconciler ClusterRole
+// which grants namespace admin
+func NSReconcilerNamespaceAdmin(opt *New) {
+	policy := rbacv1.PolicyRule{
+		APIGroups: []string{rbacv1.APIGroupAll},
+		Resources: []string{rbacv1.ResourceAll},
+		Verbs:     []string{rbacv1.VerbAll},
+	}
+	NSReconcilerCustomPolicy(policy)(opt)
+}
+
+// NSReconcilerCustomPolicy specifies a custom policy to use for the NS reconciler ClusterRole
+func NSReconcilerCustomPolicy(policy rbacv1.PolicyRule) Opt {
```

Comment:
How replacing all this with a func that accepts a vararg of rules? Then you would call `RepoSyncPolicy(policy.NamespaceAdmin(), policy.RepoSyncAdmin())` to return a `func(opt *New)`, and you could put the policy helpers in a sub-package. The nomostest packag eis already too cluttered. And I suspect it might be easier to find RepoSync- than NSReconciler- names.

Reply:
That sounds reasonable from an organization/discovery perspective. I think my current solution would be a bit more concise but I like your proposal. I'll try refactoring accordingly to see how it looks

Reply:
Updated, PTAL